src/main/scala/TraceIO.scala

```diff
@@ -4,23 +4,54 @@ import chisel3._
 import chisel3.util._
 
 import freechips.rocketchip.subsystem.{BaseSubsystem, HasTiles}
-import freechips.rocketchip.config.{Field, Config}
+import freechips.rocketchip.config.{Field, Config, Parameters}
 import freechips.rocketchip.diplomacy.{LazyModule, AddressSet, LazyModuleImp, BundleBridgeNexus}
 import freechips.rocketchip.tilelink.{TLRAM}
-import freechips.rocketchip.rocket.TracedInstruction
+import freechips.rocketchip.rocket.{TracedInstruction}
 import freechips.rocketchip.util._
+import freechips.rocketchip.tile.{BaseTile}
+import freechips.rocketchip.diplomacy.{BundleBridgeSource, BundleBroadcast, BundleBridgeNexus}
 
-case class TracePortParams(
-  print: Boolean = false
-)
-
-object TracePortKey extends Field[Option[TracePortParams]](None)
+//***************************************************************************
+// Extended Trace Instruction Utilities:
+// Used to map TracedInstructions to their Extended version and
+// used to connect the TracerV or Dromajo bridges (in FireSim and normal sim)
+//***************************************************************************
 
-case class TracedInstructionWidths(iaddr: Int, insn: Int, cause: Int, tval: Int)
+case class TracedInstructionWidths(iaddr: Int, insn: Int, wdata: Int, cause: Int, tval: Int)
 
 object TracedInstructionWidths {
+  def apply(tI: ExtendedTracedInstruction): TracedInstructionWidths =
+    TracedInstructionWidths(tI.iaddr.getWidth, tI.insn.getWidth, tI.wdata.getWidth, tI.cause.getWidth, tI.tval.getWidth)
+
+  // note: the wdata is not 0 here since we can't deal with 0 width wires
   def apply(tI: TracedInstruction): TracedInstructionWidths =
```

Comment:
[I think] We should be able to deal with an `Option` though, so i'd encourage you to use that in a follow up PR.

Reply:
Huh. I didn't know that. Ill update this after it gets in.